x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/models_management/models_list.tsx

```diff
@@ -66,22 +68,47 @@ export const ModelsList: FC = () => {
   const { toasts } = useNotifications();
 
   const [searchQueryText, setSearchQueryText] = useState('');
-
-  const [pageIndex, setPageIndex] = useState(0);
-  const [pageSize, setPageSize] = useState(10);
-  const [sortField, setSortField] = useState<string>(ModelsTableToConfigMapping.id);
-  const [sortDirection, setSortDirection] = useState<Direction>('asc');
-
+  const [filteredModels, setFilteredModels] = useState<{
+    active: boolean;
+    items: ModelItem[];
+  }>({
+    active: false,
+    items: [],
+  });
   const [isLoading, setIsLoading] = useState(false);
   const [items, setItems] = useState<ModelItem[]>([]);
   const [selectedModels, setSelectedModels] = useState<ModelItem[]>([]);
-
   const [modelsToDelete, setModelsToDelete] = useState<ModelItemFull[]>([]);
-
   const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<Record<string, JSX.Element>>(
     {}
   );
 
+  const setQueryClauses = (queryClauses: any) => {
```

Comment:
`setQueryClauses` seems like a misleading name. The function actually updates the filtered models' list. 
Also, I'm not sure if we need the `active` flag. We have the original collection of models that has been fetched and the other collection based on UI filters. I think it'd be easier to follow if the EUI table always operates on the filtered collection. In case there are no filters, you can just reset it to the original model list.

Reply:
Good point - renamed and simplified the filtered items as suggested 5b160352ccb58bc3649269931f7917bd0316bceb